deblur/test/test_workflow.py

```diff
@@ -0,0 +1,832 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2015, The Deblur Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from unittest import TestCase, main
+from shutil import rmtree
+from tempfile import mkstemp, mkdtemp
+from os import close
+
+from skbio.util import remove_files
+
+
+class workflowTests(TestCase):
+    """ Test deblur pipeline and individual methods
+        functionality """
+
+    def setUp(self):
+        """ Create working directory and two FASTA input
+            files corresponding to two samples (s1 and s2).
+            Each input file contains 120 sequences, of which
+            100 are 16S amplicons (ART simulator), 10 are
+            chimeric sequences (Grinder) and 10 are PhiX
+            artifacts (ART). The 100 amplicon sequences
+            intend to evenly represent a community of 10
+            species.
+        """
+        # test output can be written to this directory
+        self.working_dir = mkdtemp()
+
+        # create temp FASTA file for s1 sequences
+        f, self.seqs_s1_fp = mkstemp(prefix='seqs_s1_',
+                                     suffix='.fasta')
+        close(f)
+        with open(self.seqs_s1_fp, 'w') as t:
+            t.write(seqs_s1)
+        t.close()
```

Comment:
You don't need the close, the `with` statement closes it automatically. In fact, I think that it should fail, and if not, is a variable leakage....  Same for line 48